src/components/structures/auth/Login.js

```diff
@@ -265,7 +265,10 @@ module.exports = React.createClass({
     },
 
     onUsernameBlur: function(username) {
-        this.setState({ username: username });
+        this.setState({
```

Comment:
also if you didn't change the username?

Reply:
Ah, good call, I'll skip if they are the same.

Reply:
Oh wait, we don't actually know for sure if they changed here, since those should come through the change event separately... I thinks it's okay as it was. It will always clear discovery errors, and retry on blur if needed.

We could avoiding retrying on repeated blur for the same username by tracking separately that it hasn't actually changed, but it's unclear how common that actually would be.